salsa_auth/views.py

```diff
@@ -47,22 +47,39 @@ class SignUpForm(JSONFormResponseMixin, FormView):
     template_name = 'signup.html'
 
     def form_valid(self, form):
-        user = self._make_user(form.cleaned_data)
+        email = form.cleaned_data['email']
 
-        # TO-DO: Potentially intercept SMTP error for undeliverable mail here
-        self._send_verification_email(user)
+        # If the email already exists in Salsa, re-verification is not required.
+        # Authenticate the user.
+        if salsa_client.get_supporter(email):
+            return redirect('salsa_auth:authenticate')
 
-        messages.add_message(self.request,
-                             messages.INFO,
-                             'Thanks for signing up!',
-                             extra_tags='font-weight-bold')
+        # If the given email is in Salsa, and that email has hard-bounced,
+        # inform the user, and do not send another email to the invalid address.
+        if salsa_client.get_supporter(email, allow_invalid=True):
+            message_title = "Sorry we missed you!"
```

Comment:
I think having a more informative title here would be helpful so they can take action. Is there any action they can take when their status is 'HARD BOUNCE'? Maybe change this to "We can't verify this email address"